inc/class-wpseo-image-utils.php

```diff
@@ -0,0 +1,270 @@
+	 * @return array|false Image array with metadata on success, false on failure.
+	 */
+	public static function check_image_by_size( $attachment_id, $size ) {
+		/**
+		 * Filter: 'wpseo_image_image_weight_limit' - Determines what the maximum weight (in bytes) of an image is allowed to be, default is 2 MB.
+		 *
+		 * @api int - The maximum weight (in bytes) of an image.
+		 */
+		$max_size = apply_filters( 'wpseo_image_image_weight_limit', ( 2 * 1024 * 1024 ) );
+
+		$image = self::get_image( $attachment_id, $size );
+
+		if ( $image === false || ! isset( $image['path'] ) ) {
+			return $image;
+		}
+
+		$file_size = self::get_image_file_size( $image );
+		if ( $file_size > $max_size ) {
+			return false;
+		}
+
+		return $image;
+	}
+
+	/**
+	 * Find the right version of an image based on size.
+	 *
+	 * @param int    $attachment_id Attachment ID.
+	 * @param string $size          Size name.
+	 *
+	 * @return array|false Returns an array with image data on success, false on failure.
+	 */
+	private static function get_image( $attachment_id, $size ) {
+		if ( $size === 'full' ) {
+			$image         = wp_get_attachment_metadata( $attachment_id );
+			$image['url']  = wp_get_attachment_image_url( $attachment_id, 'full' );
+			$image['path'] = get_attached_file( $attachment_id );
+		}
+
+		if ( ! isset( $image ) ) {
+			$image = image_get_intermediate_size( $attachment_id, $size );
+		}
+
+		if ( ! $image ) {
+			return false;
+		}
+
+		$image['size'] = $size;
+		return self::complete_image_data( $image, $attachment_id );
+	}
+
+	/**
+	 * Finds the full file path for a given image file.
+	 *
+	 * @param string $path The relative file path.
+	 *
+	 * @return string The full file path.
+	 */
+	public static function get_full_file_path( $path ) {
+		static $uploads;
+
+		if ( ! isset( $uploads ) ) {
+			$uploads = wp_get_upload_dir();
+		}
+
+		if ( empty( $uploads['error'] ) ) {
+			return $uploads['basedir'] . "/$path";
+		}
+
+		return $path;
+	}
+
+	/**
+	 * Get the relative path of the image.
+	 *
+	 * @param string $img Image URL.
+	 *
+	 * @return string The expanded image URL.
+	 */
+	public static function get_relative_path( $img ) {
+		if ( $img[0] !== '/' ) {
+			return $img;
+		}
+
+		// If it's a relative URL, it's relative to the domain, not necessarily to the WordPress install, we
+		// want to preserve domain name and URL scheme (http / https) though.
+		$parsed_url = wp_parse_url( home_url() );
+		$img        = $parsed_url['scheme'] . '://' . $parsed_url['host'] . $img;
+
+		return $img;
+	}
+
+	/**
+	 * Get the image file size.
+	 *
+	 * @param array $image An image array object.
+	 *
+	 * @return int The file size in bytes.
+	 */
+	public static function get_image_file_size( $image ) {
+		if ( isset( $image['filesize'] ) ) {
+			return $image['filesize'];
+		}
+		// If the file size for the file is over our limit, we're going to go for a smaller version.
+		// phpcs:ignore -- If file size doesn't properly return, we'll not fail.
```

Comment:
Please mention the sniff you want PHPCS to ignore. Now it is a global ignore and all sniffs are disabled for the next line.